json/targeting.yaml

```diff
@@ -0,0 +1,411 @@
+        $ref: "#/$defs/associativeOp"
+      "*":
+        title: Multiplication Operation
+        description: Multiplication; associative, will accept and unlimited amount
+          of arguments.
+        $ref: "#/$defs/associativeOp"
+  unaryOp:
+    type: array
+    minItems: 1
+    maxItems: 1
+    items:
+      $ref: "#/$defs/args"
+  unaryRule:
+    title: Unary Operation
+    description: Any primitive JSONLogic operation with 1 operands.
+    type: object
+    additionalProperties: false
+    properties:
+      "!":
+        title: Negation Operation
+        description: Logical negation (“not”). Takes just one argument.
+        $ref: "#/$defs/unaryOp"
+      "!!":
+        title: Double Negation Operation
+        description: Double negation, or 'cast to a boolean'. Takes a single argument.
+        $ref: "#/$defs/unaryOp"
+      max:
+        title: Maximum Operation
+        description: Return the maximum from a list of values.
+        $ref: "#/$defs/unaryOp"
+      min:
+        title: Minimum Operation
+        description: Return the minimum from a list of values.
+        $ref: "#/$defs/unaryOp"
+  variadicOp:
+    type: array
+    minItems: 1
+    items:
+      $ref: "#/$defs/args"
+  variadicRule:
+    $comment: "note < and <= can be used with up to 3 ops (between)"
+    type: object
+    additionalProperties: false
+    properties:
+      or:
+        title: Or Operation
+        description: Simple boolean test, with 1 or more arguments. At a more sophisticated
+          level, "or" returns the first truthy argument, or the last argument.
+        $ref: "#/$defs/variadicOp"
+      and:
+        title: ''
+        description: Simple boolean test, with 1 or more arguments. At a more sophisticated
+          level, "and" returns the first falsy argument, or the last argument.
+        $ref: "#/$defs/variadicOp"
+      "<":
+        title: Less-Than/Between Operation. Can be used to test that one value is
+          between two others.
+        description: ''
+        $ref: "#/$defs/variadicOp"
+      "<=":
+        title: Less-Than-Or-Equal-To/Between Operation. Can be used to test that one
+          value is between two others.
+        description: ''
+        $ref: "#/$defs/variadicOp"
+      "-":
+        title: Subtraction Operation
+        $ref: "#/$defs/variadicOp"
+      merge:
+        title: Subtraction Operation
+        description: Takes one or more arrays, and merges them into one array. If
+          arguments aren't arrays, they get cast to arrays.
+        $ref: "#/$defs/variadicOp"
+      cat:
+        title: Concatenate Operation
+        description: Concatenate all the supplied arguments. Note that this is not
+          a join or implode operation, there is no “glue” string.
+        $ref: "#/$defs/variadicOp"
+  stringCompareArg:
+    oneOf:
+    - type: string
+    - $ref: "#/$defs/anyRule"
+  stringCompareArgs:
+    type: array
+    minItems: 2
```

Comment:
we can also add `"maxItems": 2` as we define the spec of str_comp arguments in our spec [1^]


[^1]: https://github.com/open-feature/flagd/blob/main/docs/reference/custom-operations/string-comparison-operation.md

Reply:
good call, added.